src/java/org/apache/cassandra/db/commitlog/CommitLogSegmentManagerCDC.java

```diff
@@ -295,18 +334,23 @@ void processNewSegment(CommitLogSegment segment)
 
         void processDiscardedSegment(CommitLogSegment segment)
         {
-            // See synchronization in CommitLogSegment.setCDCState
-            synchronized(segment.cdcStateLock)
+            if (!segment.getCDCFile().exists())
+            {
+                logger.debug("Skip updating size. The CDC commit log segement has been deleted already.");
```

Comment:
Recommend revise this to something like "Not processing discarded CommitLogSegment {}; this segment appears to have been deleted already.", segment)